src/tests/features/zone_priority.at

```diff
@@ -1,6 +1,22 @@
 FWD_START_TEST([zone - priority])
 AT_KEYWORDS(zone)
 
+m4_if(nftables, FIREWALL_BACKEND, [
```

Comment:
Please lift this out to a macro in `src/tests/functions.at`. Call it something like `SKIP_IF_FW_IN_CONTAINER`.

Please keep the modifications to the tests as minimal as possible.

Reply:
I'd really like to minimize the changes to the tests. I want to see only `SKIP_IF_FW_CONTAINER` at the start of the test. If you want the comments then they should be in the macro.

I'm not concerned about skipping `iptables`. It's not the default backend and is still covered by other testing/ci.

Reply:
`SKIP_IF_FW_CONTAINER` already  has a general comment, saying that it's skipped because the test is somehow not suitable in fw-in-container environment. But the exact reason why, depends on the test. And that can only be a comment in the test.

The comment explains the test-specific reason why the test is skipped. That cannot move to functions.at.

It takes time to investigate why a test fails, to identify that it fails for good reasons, and that it should be skipped. Those reasons are non-obvious, and if they are not recorded in a comment, it's hard to find out afterwards. Well, I'd have to dig through the git history that added the SKIP, like I did for commit "test(pr323): add comment about why CHECK_MODULE_PROTO_GRE is skipped". But that costs time, repeatedly. It's not an improvement to omit a useful comment.

We should not skip a perfectly passing iptables test, only because it would be too cumbersome to add an `IF(nftables)`. If an iptables test fails, the first thing I will do is reproduce it in fw-in-container. It's not helpful, if the test would be skipped in the debugging environment (although it is supposed to pass).

Reply:
Please lift the specific reasons out to a macro. Many of these are the same, e.g. socket buffer. These are workarounds for a specific test environment. I'd rather put the complexity in the macro so the test case reads easier. Please call the macro at the start of the tests like other `CHECK_*` macros.

Reply:
updated. How about now?